hudi-spark/src/main/java/org/apache/hudi/DataSourceUtils.java

```diff
@@ -129,6 +132,25 @@ public static AbstractHoodieDateTimeParser createDateTimeParser(TypedProperties
     }
   }
 
+  /**
+   * Create a UserDefinedBulkInsertPartitionerRows class via reflection,
+   * <br>
+   * if the class name of UserDefinedBulkInsertPartitioner is configured through the HoodieWriteConfig.
+   *
+   * @see HoodieWriteConfig#getUserDefinedBulkInsertPartitionerClass()
+   */
+  public static Option<BulkInsertPartitioner<Dataset<Row>>> createUserDefinedBulkInsertPartitionerWithRows(HoodieWriteConfig config)
+      throws HoodieException {
+    String bulkInsertPartitionerClass = config.getUserDefinedBulkInsertPartitionerClass();
```

Comment:
I have not introduced a new config for this, but reusing the same used for BulkInsertPartitioner<HoodieRecord>. 
Open to adding a new config if required.